### DISCUSSION TITLE
NimBLE

### DISCUSSION THREAD

**Discussion by er-bsman** [h2zero/NimBLE-Arduino#459] [Ideas]
I would like to get all the services and the characteristics of the Beacon? Is there any example for that to do?

**Comment by h2zero** [comment]
Hello, the function to do this is `NimBLEClient::discoverAttributes()`, just call that from your client instance after connecting.

**Reply by er-bsman** [reply]
Ok, i will try it. Also, is this function can be used for listing all the services and their characteristics?

**Reply by h2zero** [reply]
If you want to print them out to Serial that is something you'll need to do in your application.

**Comment by noakcn** [comment]
你发的邮件我已经收到了。
                                    以上为自动回复邮件
